### PR TITLE
Add some tests for parsing integers

### PR DESCRIPTION
**Pull request by hughbe**
- Currencies (positive and negative)
- Formats with length > 1
- Formats with length 0
- Formats with properties that are integers
- Double names

/cc @stephentoub

### CODE REVIEW

File: src/System.Runtime/tests/System/Int32Tests.cs [hunk 1]
```diff
@@ -179,6 +182,20 @@ public static IEnumerable<object[]> Parse_Valid_TestData()
...
             yield return new object[] { "1000", NumberStyles.Currency, currencyFormat, 1000 };
             yield return new object[] { "$(1000)", NumberStyles.Currency, currencyFormat, -1000};
+            yield return new object[] { "($1000)", NumberStyles.Currency, currencyFormat, -1000 };
+            yield return new object[] { "$-1000", NumberStyles.Currency, currencyFormat, -1000 };
+            yield return new object[] { "-$1000", NumberStyles.Currency, currencyFormat, -1000 };
+
+            NumberFormatInfo emptyCurrencyFormat = new NumberFormatInfo() { CurrencySymbol = "" };
+            yield return new object[] { "100", NumberStyles.Currency, emptyCurrencyFormat, 100 };
+
+            // If CurrencySymbol and Negative  are the same, NegativeSign is preferred
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/Int32Tests.cs]: Nit: extra space between "Negative" and "are"

File: src/System.Runtime/tests/System/Int32Tests.cs [hunk 2]
```diff
@@ -345,6 +392,24 @@ public static IEnumerable<object[]> Parse_Invalid_TestData()
             NumberFormatInfo decimalFormat = new NumberFormatInfo() { NumberDecimalSeparator = "." };
             yield return new object[] { "67.90", NumberStyles.AllowDecimalPoint, null, typeof(OverflowException) };
 
+            // Parsing integers doesn't allow NaN, PositiveInfinity or NegativeInfinity
+            NumberFormatInfo doubleFormat = new NumberFormatInfo()
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/Int32Tests.cs]: Where is this used?

File: src/System.Runtime/tests/System/Int32Tests.cs (hunk 1 above)
Comment (hughbe) [inline src/System.Runtime/tests/System/Int32Tests.cs]: fixed

File: src/System.Runtime/tests/System/Int32Tests.cs (hunk 2 above)
Comment (hughbe) [inline src/System.Runtime/tests/System/Int32Tests.cs]: good catch - fixed

**Comment by hughbe** [comment]
Test  Innerloop OSX Debug Build and Test
Test  Innerloop OSX Release Build and Test

**Comment by hughbe** [comment]
Test Innerloop OSX Debug Build and Test
Test Innerloop OSX Release Build and Test

**Comment by stephentoub** [comment]
@mmitche, all of the OSX runs still appear to be failing at the end:
```
[osx_debug_prtest@2] $ /bin/sh -xe /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/hudson7926712879264428998.sh
+ tar -czf bin/build.tar.gz bin/AnyOS.AnyCPU.Debug bin/Linux.AnyCPU.Debug bin/OSX.AnyCPU.Debug bin/OSX.x64.Debug bin/Unix.AnyCPU.Debug bin/Windows_NT.AnyCPU.Debug bin/ref bin/packages --exclude '*.Tests'
tar: --exclude: Cannot stat: No such file or directory
tar: *.Tests: Cannot stat: No such file or directory
tar: Error exit delayed from previous errors.
```

**Comment by stephentoub** [comment]
LGTM.  Thanks, @hughbe.

**Pull request merged**